docs/proposals/docker-registry.md

```diff
@@ -0,0 +1,36 @@
+## Improve docker-registry
+
+We need to stop using `linkedBlobStore` from `docker/distribution`. The main purpose for this module is to store information
+about blobs because `docker/distribution` don't have own database. It places information in the filesystem.
+
+Openshift already uses a etcd as a database. All infomation about images/imagestreams stored there. We can hold information
+about the blobs in repository in the database as well. In this case we need store only the blobs's payload in the filesystem
+using our own objects layout. In this case, we get complete control over the blobs.
+
+To do this we will need to implement part of the functionality of the `linkedBlobStore`. We can create our own middleware for
+repository and storage driver. The middleware for storage driver will be used for reading and writing in old (docker/distribution)
+and new layout.
+
+### Benefits
+
+* we will be able to use different storage drivers (filesystem, s3, swift, etc.) to store blobs;
+* we will get complete control over the blobs;
+* we will be able to gradually migrate.
+
+### Negative aspects
+
+* We have to implement part of the functionality of the `blobStore` and `linkedBlobStore`.
+
+### New objects layout
+
+In the layout I propose to use are similar to `docker/distribution` but simpler:
```

Comment:
Exactly. You could also link to the pieces of code we'd need to export. Also add a note that we already tried to negotiate with upstream about exporting it.